service/src/main/java/uk/gov/hmcts/reform/fpl/model/DirectionResponse.java

```diff
@@ -21,4 +22,51 @@
     private final String cannotComplyReason;
     private final List<String> c2Uploaded;
     private final DocumentReference cannotComplyFile;
+    @JsonIgnore
+    private String respondingOnBehalfOf;
+
+    @JsonGetter("respondingOnBehalfOfRespondent")
+    private String getRespondingOnBehalfOfRespondent() {
+        if (respondingOnBehalfOf != null && respondingOnBehalfOf.contains("RESPONDENT")) {
```

Comment:
These could be refactored into private method reused in getters, e.g.

```
getValueIfContainsOrNull("RESPONDENT")
```

Similarly same could be done for setters e.g.:

```
setValueIfNotNull(value);
```

Next step could be consist extracting methods used by Jackson to MixIn class but it might be too much hassle compared to all other rubbish we had to do.

Reply:
I have refactored to your suggested methods, cuts down code. The next step I think I am fine to accept as a next step, but I see no reason for it to happen within the scope of this story given the strict deadlines in place 👍